### PR TITLE
en-gb style for elipsis

### PR DESCRIPTION
**Pull request by brianteeman**
simple update to add a space before the ... as per the style guide

### CODE REVIEW

**Comment by zero-24** [comment]
RTC Thanks :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7386">issues.joomla.org/joomla-cms/7386</a>.</sub>

**Comment by infograf768** [comment]
In these cases, there is no reason to add a space before the ellipsis as it is a real end of phrase/expression ellipsis and not used as a replacement for some content as in [ ... ] where adding a space before and after do make sense.
Example:
JGLOBAL_SHOW_FULL_DESCRIPTION="Show full description ..."
is wrong imho
it should be changed back to
JGLOBAL_SHOW_FULL_DESCRIPTION="Show full description..."

Same for the Read More...

**Comment by brianteeman** [comment]
Sorry you are wrong.

**Comment by infograf768** [comment]
>Unless it is at the end of a string where there is no space or period after. 

That is clear enough imho and common to many languages. It is such to simply avoid the ellipsis (whether as dots or as hellip) to jump to next line when text is justified as the normal space before would be a "breaking space". Also an ellipsis never has a dot after the 3 dots.

**Comment by Kubik-Rubik** [comment]
Could you please decide what we will use, so that we can merge or close this PR?

**Comment by brianteeman** [comment]
What is to decide. This pr follows the style guide.

**Comment by Bakual** [comment]
Where did JM cite from?

**Comment by brianteeman** [comment]
No idea which quote you refer to but this PR follows the style guide for two strings that were missed

**Comment by brianteeman** [comment]
Doh. The second correction has a space it shouldn't as its a new sentence. Will update when I get home

**Comment by brianteeman** [comment]
fixed it - my eyes failed me on that space or which issue JM referred to

**Comment by Bakual** [comment]
> "An ellipsis is a set of three periods ( ... ) indicating an omission. The group of periods should have a single space on either side. Unless it is at the end of a string where there is no space or period after."

That looked to me like a quote and I wondered where it is from.

> No idea which quote you refer to but this PR follows the style guide for two strings that were missed

Yeah, I get that. But I hope the style guide isn't set in stone when some other non-Joomla (?) style guide would suggest otherwise. That's why I would love to know where JM got that quote from.

**Comment by wojsmol** [comment]
@Bakual https://github.com/joomla/user-interface-text/blob/master/punctuation.md#ellipsis

**Comment by brianteeman** [comment]
That quote is from the style guide.

I made a mistake in the Pr which JM was indicating but as there was no line
reference I missed it.

The style guide if you actually read it is not a figment of my imagination
but a result of considerable research and all references are included.

I am very disappointed you think so little of it and all the hard work that
went in to making it. When contributions are not valued it makes a
contributor feel its not worth bothering.

**Comment by wojsmol** [comment]
@brianteeman If your answer was addressed to me it is very sorry that so briefly answered the @Bakual question. I did not want to offend anyone this way.

**Comment by brianteeman** [comment]
@wojsmol it wasn't. Sorry if you thought it was.

**Comment by Bakual** [comment]
> The style guide if you actually read it is not a figment of my imagination
but a result of considerable research and all references are included.

> I am very disappointed you think so little of it and all the hard work that
went in to making it. When contributions are not valued it makes a
contributor feel its not worth bothering. 

I was under the impression that JM had quoted a styleguide which indicated something else than our own styleguide.
I don't know what makes you think I do not value it. I haven't said any such thing and it is a false assumption. I just asked for the source of the quote to put it into context, that's all.

Since that is sorted now, and you both actually have found consent, there is nothing left stopping this PR.

**Comment by infograf768** [comment]
`<rant>`
That guide is plain wrong and this PR too. And I explained why.
Anybody who has done a bit of serious typography, in any European language, will explain to you that such a punctuation <b>should never jump to next line</b> but should stay attached to the portion of phrase it is used in. It should never be separated from the word before <b> except when it replaces a portion of phrase inside a phrase</b>. Using a "normal space", instead of nothing or a non-breaking space, before the ellipsis may get that one jump to next line.

Examples:
```
I never use a space before the ellipsis but ... some people insists on doing it.
```
May give, if the width available for the paragraph forces the ellipsis to jump to next line.
```
I never use a space before the ellipsis but 
... some people insists on doing it.
```
Or
```
I never use a space before the ellipsis but ...some people insists on doing it.
```
may give
```
I never use a space before the ellipsis but 
...some people insists on doing it.
```
It should always give:
```
I never use a space before the ellipsis but... 
some people insists on doing it.
```
Or, when placed at the end of a phrase or expression, we should always get 
```
That is a very long story to tell...
```
and <b>never</b>
```
That is a very long story to tell
...
```
In
```
-JGLOBAL_SHOW_READMORE_DESC="If set to Show, the Read more ... Link will show if Main text has been provided for the Article."
+JGLOBAL_SHOW_READMORE_DESC="If set to Show, the Read more ...Link will show if Main text has been provided for the Article."
 ```
It is even worse... as it is quoting an expression which <b>does contain the ellipsis</b>. Not counting the useless Uppercase used for `Link` and `Article`, it is obvious that the ellipsis concerns the `Read more...` and not a type of expression similar to my `It is even worse...` above.
The value should be:
`"If set to Show, the 'Read more...' link will show if Main text has been provided for the article."`

It is only when it replaces part of a phrase or paragraph that it "could" have spaces before or after, <b>if it is not surrounded by brackets or parenthesis, which is much better</b>. In these cases one does not care if it is forced to jump to next line if necessary. 
Example when quoting.
Churchill wrote in <i>The birth of Britain</i>:
```
 “The gift which Roman civilization had to bestow was civic and political. 
[...] It was a period of hope.”
```
The `[...]` replaces the content between the two phrases: 
`Towns were planned in chessboard squares for communities dwelling under orderly government. The buildings rose in accordance with the pattern standardized throughout the Roman world. Each was complete with its forum, temples, courts of justice, gaols, baths, markets, and main drains. During the first century the builders evidently took a sanguine view of the resources and future of Britannia, and all their towns were projected to meet increasing population.`

In conclusion, the ellipsis should be treated like most punctuation marks. One would not add a space before a dot, nor take the space off after it.
Typography is a serious business and being raised with a specific language does not make one aware of its subtilities by birth.
`</rant>`

**Comment by brianteeman** [comment]
Once again you are wrong. It doesn't matter how many words you use it
doesn't make you correct.

Once again this is not something I made up. Check the references in the
guide.

Perhaps you prefer this one as it is NOT British
European Commission Directorate-General for Translation
http://ec.europa.eu/translation/english/guidelines/documents/styleguide_english_dgt_en.pdf

**Comment by infograf768** [comment]
Indeed this doc states there should be a space before the ellipsis in English.

But I suggest you re-read it and specially the use of a "hard space" which is also known as a "non-breaking space" `before` the ellipsis ( https://en.wikipedia.org/wiki/Hard_space ). 
The doc also `never` says that there should be no space after the ellipsis (except evidently at the end of a phrase.

Excerpt from http://ec.europa.eu/translation/english/guidelines/documents/styleguide_english_dgt_en.pdf

>When replacing one or more words in the middle of a sentence, it is preceded by a hard space and followed by a normal space.
When replacing one or more words at the end of a sentence, it is preceded by a hard space.

The reason for the "hard space" is exactly the one I described above, i.e. to forbid an automatic line breaking. 
In this case please correct your PR (and your guide) and use the `&nbsp;` space before and also never use the ellipsis without a normal space after it, as I quoted above 
`JGLOBAL_SHOW_READMORE_DESC="If set to Show, the Read more ...Link will show if Main text has been provided for the Article."`
which is plain wrong.
It should be, reading at the doc
`JGLOBAL_SHOW_READMORE_DESC="If set to Show, the 'Read more&nbsp;...' link will show if Main text has been provided for the Article."`

**Comment by Bakual** [comment]
I think the hard spaces (non-breaking) would make sense for the reasons JM outlined.
Can we add this to out styleguide which would then be in line with the european one? And adjust the strings changed in this PR?

And please all, keep your language a bit more civil. I know you are not best friends but this is a public place. It's not the place for insulting others.

**Comment by brianteeman** [comment]
Please note this PR brought all uses of ... In line with the rest of the language bstrings. This change is for consistency throughout the cms which is the number one aim of the style guide.

If you want something else then do it yourself. You clearly place no value on contributions. I will close the working group and remove the style guide. Bye

**Comment by Bakual** [comment]
So the style guide is written in stone and deemed bug free? That would be the first thing in Joomla which has aquired that badge. I can't believe that this is the case.

Don't you agree that JM has some valid points and that the style guide you linked yourself mentions those hard spaces?
I think it should be possible to discuss improvements to the style guide when there are valid points raised. Maybe this place is the wrong place anmd it should be better done in the style guide repo. Would you prefer that?
